app/routes/tasks.js

```diff
@@ -5,11 +5,23 @@ const API_BASE_URL = ENV.BASE_API_URL;
 
 export default class TasksRoute extends Route {
   model = async () => {
-    const response = await fetch(`${API_BASE_URL}/tasks/self`, {
-      credentials: 'include',
-    });
-    const allTasks = await response.json();
-
-    return allTasks;
+    try {
+      const response = await fetch(`${API_BASE_URL}/tasks/self`, {
+        credentials: 'include',
+      });
+      if (!response.ok) {
+        if (response.status == 401) {
+          throw new Error('Please log in to continue');
+        }
+        throw new Error('Oops, We ran into a problem!');
+      }
+      return await response.json();
+    } catch (error) {
+      alert(error);
+      window.open(
+        'https://github.com/login?client_id=23c78f66ab7964e5ef97&return_to=%2Flogin%2Foauth%2Fauthorize%3Fclient_id%3D23c78f66ab7964e5ef97',
```

Comment:
Let's use this URL: https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97

Reply:
Sure!